spectrochempy/widgets/baselinecorrector.py

```diff
@@ -0,0 +1,311 @@
+import ipywidgets as widgets
+
+from IPython.display import display
+
+from spectrochempy.core.dataset.nddataset import NDDataset
+from spectrochempy.core.processors.concatenate import concatenate
+from spectrochempy.core.processors.baseline import BaselineCorrection
+from spectrochempy.core.readers.importer import read
+from spectrochempy.core.plotters.multiplot import multiplot
+from spectrochempy.core import error_
+
+
+__all__ = ["BaselineCorrector"]
+
+
+class BaselineCorrector:
+    """
+    Launch a GUI for baseline corrections.
+
+    Wrapper of scp.BaselineCorrection(X), with widgets for dataset slicing,
+    input parameters and graphical output.
+    Should be run in jupyter notebook (not jupyter lab) with the widget backend
+    (magic `%matplotlib widget`).
+
+    Parameters
+    ----------
+    X : NDDataset, default: None
+        The NDDataset to process. If None, a FileUpload widget is enabled.
+    initial_ranges : list, optional, default: None
+        The initial regions where to compute the baseline. If not given, 10% on each
+        side of the spectra will be taken as a starting range's list.
+
+
+    Attributes
+    ----------
+    original : NDDataset
+        The (sliced) NDDataset, without baseline correction
+    corrected : NDDataset
+        The baseline corrected NDDataset
+    baseline : NDDataset
+        The baselines
+
+    See Also
+    --------
+    BaselineCorrection : Baseline correction method.
+
+    Notes
+    -----
+    The `BaselineCorrector()` widget can be used in jupyter notebooks.
+    - The GUI buttons are as follows:
+        - `upload`: upload files (disabled if a NDDataset is passed as parameter).
+          Uploading file will not trigger the reading
```

Comment:
I am wondering what is the reason of not triggering the processing

Reply:
because I was lazy :-). Now it also triggers reading/processing (see last sereis of commits [06da720](https://github.com/spectrochempy/spectrochempy/pull/397/commits/06da720a32e3aba7ce13399c897f11583fef6d7c) - [a3b873c](https://github.com/spectrochempy/spectrochempy/pull/397/commits/a3b873c2744f77fbe39841b69e4ab7240751d8da)). 
With this change, it still works in jupyter notebook, but not in jupyter lab (plots are visible in the log, though ;   so I don't know how to debug this ).

Reply:
I will try to look at this this afternoon.